src/mdoc/model/DeviceResponse.ts

```diff
@@ -122,19 +109,32 @@ export class DeviceResponse {
     verifierGeneratedNonce: string,
   ): DeviceResponse {
     this.usingSessionTranscriptBytes(
-      cborEncode(
-        DataItem.fromData([
-          null, // deviceEngagementBytes
-          null, // eReaderKeyBytes
-          [mdocGeneratedNonce, clientId, responseUri, verifierGeneratedNonce],
-        ]),
-      ),
+      this.#oid4vptranscript(mdocGeneratedNonce, clientId, responseUri, verifierGeneratedNonce),
     );
     return this;
   }
 
+  async #oid4vptranscript(
```

Comment:
minor: consider moving this helper (and the one for web api) to utils.ts (just to avoid duplicating code)